test/tests/functional/pbs_ralter.py

```diff
@@ -2842,3 +2842,42 @@ def test_restart_revert(self):
         self.server.expect(RESV, attrs, id=rid)
         wait = start - time.time()
         self.check_resv_running(rid, offset=wait)
+
+    def test_alter_degrade_reconfirm_standing(self):
+        """
+        Test that if a standing reservation is altered, degraded,
+        then reconfirmed, the reservation will use the original
+        select
+        """
+        duration = 60
+        offset = 60
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'reserve_retry_time': 2})
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, standing=True, select="2:ncpus=2")
+
+        confirmed = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, confirmed, id=rid)
+
+        self.server.alterresv(rid, {ATTR_l: "select=1:ncpus=2"})
+        self.server.expect(RESV, confirmed, id=rid)
```

Comment:
Why call out to alterresv() directly and not use alter_a_reservation()?  It handles making sure it's confirmed properly.

Reply:
because i didn't see alter_a_reservation needs to pass start and end time, i thought it would just be easier to alter the reservation without the helper function.

Reply:
You have the start and end from submit_and_confirm_reservation().  I'd use the helper function.  It does a lot of stuff that is good to make sure the alter worked.